agent/loop.py

```
@@ -619,6 +619,20 @@ async def run(
+                # can reuse proven patterns when planning similar tasks.
+                if result.get("status") == "applied" and result.get("changed_files"):
+                    try:
+                        from agent.procedural_memory import get_procedural_memory
+                        changed = ", ".join(result["changed_files"][:3])
+                        get_procedural_memory().record_success(
+                            goal_summary=plan.goal[:120],
+                            step_description=step_data.get("description", "")[:200],
+                            action_summary=f"edited {changed}",
+                        )
+                    except Exception:
+                        pass
```

Ruff (0.16.1)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)